### DISCUSSION TITLE
Plugin protected pages

### DISCUSSION THREAD

**Discussion by franckbertuzzi** [GetPublii/Publii#1494] [Ideas]
Hello
What do you think about a plugin allowing to protect access (with a password) to certain articles? This would be very useful for theme docs : for exemple, when you want ro restrict access to a confidential content.
Thank you

**Comment by candidexmedia** [comment]
That would be useful, yes!

I know that there is already a paywall plug-in in the works: https://p4p.pages.dev/plugin-paywall/

**Reply by internettips** [reply]
@candidexmedia 

Isn't the Paywall plugin at: https://p4p.pages.dev/plugin-paywall/ still a 'work in progress'? There doesn't seem to be any (easy) way to obtain the plugin yet.

**Reply by candidexmedia** [reply]
@internettips That is why I wrote "in the works" 😉

You could always reach out to the dev via their [contact form](https://p4p.pages.dev/contact/), or by mentioning their handle to ask about testing it or how development is going. It looks like it might require purchasing a license once it's ready.

![](https://i.imgur.com/u8ls6Wn.png)

**Reply by internettips** [reply]
@candidexmedia 

Thanks for your feedback. The download link in your screenshot isn't of course a live / active download link -- it's more illustrative of the potential I guess. There doesn't seem to be any obvious way to register on the site yet, etc. Have tried contacting -- though he sounds like someone who is really busy, so I didn't want to pester too much. No problem in waiting to discover if his plugin licence becomes available to buy at some point. At first glance though, it looks quite interesting; he's done some amazing work with it. Combining Publii with a gated content solution seems like a great idea -- even though 'out of the box' Publii doesn't really seem to be designed for that. But things can evolve can't they? :-)

**Comment by 3muchfun** [comment]
I would go for it and give it a try. The programmer is a cool guy.

**Comment by franckbertuzzi** [comment]
I would love this plugin
Thank you and let me know

**Comment by jpbraun** [comment]
This seems to be just the right plug-in for password-protected pages and posts. I hope that the developer will reply to me.

**Comment by gpsblues** [comment]
Hi, 
on Apache servers, static pages can be password protected with htaccess directives and a file called htpasswd. I have already implemented this on some sites.

See https://www.mountainleaderitalia.org/area-riservata/. This solution can be valid for simple sites.

I am currently working on a plugin that facilitates this implementation, but I have encountered more difficulties than expected. 

The .htpasswd file, for example, does not seem to be transferable with Publii. Additionally, there is the problem of the hash that can probably be overcome with the use of Crypto JS.

Moreover, one must consider the differences with pretty URLs or not. Finally, there are different modes of application between one server and another.

In practice, I would have to dedicate much more time than expected. In the end, for this method, a manual implementation might be simpler.

**Comment by krzakx** [comment]
hey
what is status of plugin? I am looking for paywall solution.

**Comment by Backspaze** [comment]
I found this discussion when searching for a solution for password protecting static sites. I'd previously used https://github.com/robinmoisson/staticrypt (or https://github.com/Greenheart/pagecrypt, can't remember which) for a one-off case, and I was about to create a new discussion here about help with creating a plugin that uses any of those solutions.

But the paywall plugin mentioned in this discussion seems like it already does what those do, and more. Here's hoping that this plugin gets [added to the marketplace](https://getpublii.com/promote-your-extensions/) for easy access.
